hiplot/fetchers.py

```diff
@@ -13,31 +13,61 @@
 import os
 import importlib
 import importlib.util
-from typing import Dict, List, Optional, Callable, Any
+import typing as tp
 from pathlib import Path
 
 from . import experiment as hip
 from .fetchers_demo import README_DEMOS
 
 
 class NoFetcherFound(Exception):
-    pass
+    def __init__(self, uri: str):
+        super().__init__(f"Unable to fetch an HiPlot experiment from '{uri}'")
 
 
-def load_xp_with_fetchers(fetchers: List[hip.ExperimentFetcher], uri: str) -> hip.Experiment:
+def load_xp_with_fetchers_partial(fetchers: tp.List[hip.ExperimentFetcher], uri: str) -> tp.Tuple[hip.Experiment, int]:
+    """
+    Attempts to parse an uri.
+    Returns a corresponding experiment, and position in uri following the last used character.
+
+    Example:
+    /path1
+    /path2
+    Returns the Experiment for `/path1` and position the substring `\n/path2`
+    """
+    eol = uri.find("\n")
+    if eol == -1:
+        eol = len(uri)
     for f in fetchers:
         try:
-            return f(uri)
+            endoffetcher = eol
+            if hasattr(f, "partial"):
+                endoffetcher = f.partial(uri)  # type: ignore
+            return f(uri[:endoffetcher]), endoffetcher
         except hip.ExperimentFetcherDoesntApply:
             continue
     raise NoFetcherFound(uri)
 
 
+def load_xp_with_fetchers(fetchers: tp.List[hip.ExperimentFetcher], uri: str) -> hip.Experiment:
+    uri = uri.lstrip()
+    xps: tp.List[hip.Experiment] = []
+    while uri:
+        xp, endfetcher = load_xp_with_fetchers_partial(fetchers, uri)
+        uri = uri[endfetcher:].lstrip()
+        xps.append(xp)
+    assert xps, uri
+    return hip.Experiment.merge({
+        f"{i}": xp
+        for i, xp in enumerate(xps)
+    }) if len(xps) > 1 else xps[0]
```

Comment:
That's where we decide for the names of the experiments

Reply:
than means 0, 1, 2 etc?